src/ow1/heroes/mei.opy

```diff
@@ -15,7 +19,32 @@ rule "[mei.opy]: Initialize Mei":
 rule "[mei.opy] Freeze enemy": # credit to snappycreeper for this rule
     @Event playerDealtDamage
     @Hero mei
-    @Condition eventPlayer.isFiringPrimaryFire() == true
+    @Condition eventPlayer.isFiringPrimaryFire() == true and victim.isFrozen == false
+
+    wait(0.016, Wait.ABORT_WHEN_FALSE)
+    if victim.startFreeze == false: # start the freeze time at 20
+        victim.freeze_time = 20
+    if victim.freeze_time < 100: # make sure it doesn't reset back to 20 
+        victim.startFreeze = true
+    victim.freeze_time += 2.85 # 2.85% each tick from wiki
+    if victim.freeze_time <= 70: # make sure you can only max freeze 70% movement speed
+        victim.setMoveSpeed(victim.freeze_time)
```

Comment:
Shouldn't this be 100-freeze_time?

Because this logic is saying that the victim should move faster the more frozen they are.

Reply:
I made a mistake with setmovespeed.